app/V1Module/presenters/SubmissionsPresenter.php

```diff
@@ -56,18 +64,30 @@ class SubmissionsPresenter extends BasePresenter {
    */
   public $submissionAcl;
 
+
   /**
-   * Get a list of all submissions, ever
+   * Get information about the evaluations of a solution
    * @GET
+   * @param string $id Identifier of the solution
+   * @throws ForbiddenRequestException
    */
-  public function actionDefault() {
-    if (!$this->submissionAcl->canViewAll()) {
-      throw new ForbiddenRequestException();
+  public function actionEvaluations(string $id) {
+    $solution = $this->assignmentSolutions->findOrThrow($id);
+    if (!$this->submissionAcl->canViewDetail($solution)) {
+      throw new ForbiddenRequestException("You cannot access this solution evaluations");
     }
 
-    $submissions = array_filter($this->submissions->findAll(), (function (Submission $submission) {
-      return $this->submissionAcl->canViewDetail($submission);
-    }));
+    $submissions = array_filter($solution->getSubmissions()->getValues(),
+      function (AssignmentSolutionSubmission $submission) use ($solution) {
+        return $this->submissionAcl->canViewEvaluation($solution, $submission);
+    });
+
+    // display only proper data for logged user
```

Comment:
I would probably reword this along the lines of "Display only data that the current user can view".